packages/shopify-app-remix/src/server/authenticate/public/types.ts

```diff
@@ -1,7 +1,67 @@
 import type {AuthenticateCheckout} from './checkout/types';
 import type {AuthenticateAppProxy} from './appProxy/types';
 
+// Eventually this will be just the `{}` bit without `AuthenticateCheckout &`
+// We have this is because in v1 public WAS the only public authenticate method
+// But it became tightly coupled to authentictaing Checkout requests.
+// In V2 you will have only public.checkout() and public.appProxy(), no public()
+
+/**
+ * Methods for authenticating Requests from Shopify's public surfaces
+ *
+ * To maintain backwards compatability this is a function and an object.
+ *
+ * Do not use `authenticate.public()`. Use `authenticate.public.checkout()` instead.
+ * `authenticate.public()` will be removed in v2.
+ *
+ * Methods are:
+ *
+ * - `authenticate.public.checkout()` for authenticating requests from checkout extensions
+ * - `authenticate.public.appProxy()` for authenticating requests from app proxies
```

Comment:
I wonder if it's worth adding this - intellisense would technically fill in that gap, and it would show up in the docs. That being said, this might be something we want to add throughout the `authenticate` objects.

Reply:
Yeah that's a fair point! Let's go with this.